packages/neon/neon_talk/lib/src/pages/room.dart

```diff
@@ -121,14 +129,44 @@ class _TalkRoomPageState extends State<TalkRoomPage> {
                   final previousMessage =
                       messagesResult.requireData.length > index + 1 ? messagesResult.requireData[index + 1] : null;
 
+                  Widget child = TalkMessage(
+                    chatMessage: message,
+                    lastCommonRead: lastCommonReadSnapshot.data,
+                    previousChatMessage: previousMessage,
+                  );
+
+                  if (previousMessage != null) {
+                    final date = parseTimestampDate(message.timestamp);
+                    final previousDate = parseTimestampDate(previousMessage.timestamp);
+                    if (!date.isAtSameMomentAs(previousDate)) {
```

Comment:
This looks a bit expensive to me as it creates multiple new DateTime objects per message.
I'd prefer a method (maybe extension method) that checks whether both are the same date.

```dart
extension Test on DateTime {
bool isSameDate(DateTime other){
// converting to utc is inexpesive as all times should already be in that timezone
final thisUtc = this.toUtc();
final otherUtc = other.toUtc();

// checking in reverse as it is more likely that the previous message has a different day than year
return thisUtc.day == otherUtc.day && thisUtc.month == otherUtc.month && thisUtc.year == otherUtc.year;
}
}
```

Reply:
I have a different more primitive idea: We do an integer division of the timestamp by the number of seconds in a day and compare that value with the value for the previous date. That's a lot cheaper to compute and doesn't allocate any date objects (and also easier to understand IMO as it is a conversion and comparison in one step).